src/energyplus/ForwardTranslator/ForwardTranslateThermalZone.cpp

```diff
@@ -169,159 +169,775 @@ namespace openstudio {
+              zoneVentilation.setString(ZoneVentilation_DesignFlowRateFields::ZoneorZoneListName, tzName);
+              zoneVentilation.setString(ZoneVentilation_DesignFlowRateFields::ScheduleName, this->alwaysOnSchedule().name().get());
+              zoneVentilation.setString(ZoneVentilation_DesignFlowRateFields::DesignFlowRateCalculationMethod, "AirChanges/Hour");
+              zoneVentilation.setDouble(ZoneVentilation_DesignFlowRateFields::AirChangesperHour, outdoorAirFlowAirChangesperHour);
+              m_idfObjects.push_back(zoneVentilation);
+            }
+          }
+        }
+      }
     }
+  }
+  void ForwardTranslator::translateThermalZoneSpacesToEnergyPlusSpaces(model::ThermalZone& modelObject, IdfObject& idfObject) {
+
+    std::string tzName = modelObject.nameString();
 
-    // Spaces
-    // Note, when you reach this point of the forward translator thermalZone.combineSpaces() has already been called,
-    // This happens in ForwardTranslator::translateModelPrivate. As a result, each zone has 0 or 1 space only
     std::vector<Space> spaces = modelObject.spaces();
+
+    // INCLUDE SPACE TRANSLATION (new behavior)
     if (spaces.empty()) {
-      LOG(Warn, "ThermalZone " << modelObject.name().get() << " does not have any geometry or loads associated with it.");
+      LOG(Warn, "ThermalZone " << tzName << " does not have any geometry or loads associated with it.");
     } else {
-      OS_ASSERT(spaces.size() == 1);
-
-      idfObject.setDouble(openstudio::ZoneFields::DirectionofRelativeNorth, spaces[0].directionofRelativeNorth());
 
-      idfObject.setDouble(openstudio::ZoneFields::XOrigin, spaces[0].xOrigin());
+      // sort by space name
+      std::sort(spaces.begin(), spaces.end(), WorkspaceObjectNameLess());
+      LOG(Warn, "TEST1");
+      boost::optional<double> directionofRelativeNorth;
+      if (!spaces[0].isDirectionofRelativeNorthDefaulted()) {
+        directionofRelativeNorth = spaces[0].directionofRelativeNorth();
+      }
+      double xOrigin = spaces[0].xOrigin();
+      double yOrigin = spaces[0].yOrigin();
+      double zOrigin = spaces[0].zOrigin();
+      double sumFloorArea = 0.0;
+      double sumNumberOfPeople = 0.0;
+      double sumVolume = 0.0;
+      double totalFloorArea = 0.0;  // only area included in total floor area
+      bool needToSetFloorArea = false;
+      bool anyNotPartofTotalFloorArea = false;
+      bool partofTotalFloorArea = spaces[0].partofTotalFloorArea();
+
+      for (const Space& space : spaces) {
+
+        // if all spaces have same directionOfRelativeNorth use that, otherwise clear it
+        if (!space.isDirectionofRelativeNorthDefaulted()) {
+          if (directionofRelativeNorth && (*directionofRelativeNorth == space.directionofRelativeNorth())) {
+            // no-op
+          } else {
+            directionofRelativeNorth.reset();
+          }
+        }
 
-      idfObject.setDouble(openstudio::ZoneFields::YOrigin, spaces[0].yOrigin());
```

Comment:
Is this method of computing the origin taken from the comnbine spaces algorithm?